consensus/polybft/bridge_manager.go

```diff
@@ -0,0 +1,488 @@
+	}
+
+	return nil
+}
+
+// BuildEventRoot builds exit event root for given epoch
+func (b *bridgeManager) BuildEventRoot(epoch uint64) (types.Hash, error) {
+	return b.checkpointManager.BuildEventRoot(epoch)
+}
+
+// Commitment returns the pending signed state sync commitment
+func (b *bridgeManager) Commitment(pendingBlockNumber uint64) (*CommitmentMessageSigned, error) {
+	return b.stateSyncManager.Commitment(pendingBlockNumber)
+}
+
+// GenerateProof generates proof for a specific event type
+func (b *bridgeManager) GenerateProof(eventID uint64, pType proofType) (types.Proof, error) {
+	switch pType {
+	case StateSync:
+		return b.stateSyncManager.GetStateSyncProof(eventID)
+	case Exit:
+		return b.checkpointManager.GenerateExitProof(eventID)
+	default:
+		return types.Proof{}, fmt.Errorf("unknown proof type requested: %v", pType)
+	}
+}
+
+// PostBlockAsync is called on finalization of each block (either from consensus or syncer)
+// but it doesn't require return of any kind, and is done asynchronously
+func (b *bridgeManager) PostBlockAsync(req *PostBlockRequest) {
+	// we will do PostBlock on checkpoint manager at the end, because it only
+	// sends a checkpoint in a separate routine. It doesn't do any db operations
+	b.checkpointManager.PostBlock(req)
+}
+
+// close stops ongoing go routines in the manager
+func (b *bridgeManager) Close() {
+	b.stateSyncRelayer.Close()
+	b.exitEventRelayer.Close()
+}
+
+// initStateSyncManager initializes state sync manager
+// if bridge is not enabled, then a dummy state sync manager will be used
+func (b *bridgeManager) initStateSyncManager(
+	bridgeBackend BridgeBackend,
+	runtimeConfig *runtimeConfig,
+	logger hclog.Logger) error {
+	stateSyncManager := newStateSyncManager(
+		logger.Named("state-sync-manager"),
+		runtimeConfig.State,
+		&stateSyncConfig{
+			key:               runtimeConfig.Key,
+			dataDir:           runtimeConfig.DataDir,
+			topic:             runtimeConfig.bridgeTopic,
+			maxCommitmentSize: maxCommitmentSize,
+		},
+		bridgeBackend,
+	)
+
+	b.stateSyncManager = stateSyncManager
+
+	return b.stateSyncManager.Init()
+}
+
+// initCheckpointManager initializes checkpoint manager
+// if bridge is not enabled, then a dummy checkpoint manager will be used
+func (b *bridgeManager) initCheckpointManager(
+	eventProvider *EventProvider,
+	runtimeConfig *runtimeConfig,
+	logger hclog.Logger) error {
+	txRelayer, err := txrelayer.NewTxRelayer(
+		txrelayer.WithIPAddress(runtimeConfig.GenesisConfig.Bridge.JSONRPCEndpoint),
+		txrelayer.WithWriter(logger.StandardWriter(&hclog.StandardLoggerOptions{})))
+	if err != nil {
+		return err
+	}
+
+	b.checkpointManager = newCheckpointManager(
+		wallet.NewEcdsaSigner(runtimeConfig.Key),
+		runtimeConfig.GenesisConfig.Bridge.CheckpointManagerAddr,
+		txRelayer,
+		runtimeConfig.blockchain,
+		runtimeConfig.polybftBackend,
+		logger.Named("checkpoint_manager"),
+		runtimeConfig.State)
```

Comment:
WDYT (I have provided checkpoint_manager logger instance to `TxRelayer` `Writer`)

```suggestion
	logger = logger.Named("checkpoint_manager")
	txRelayer, err := txrelayer.NewTxRelayer(
		txrelayer.WithIPAddress(runtimeConfig.GenesisConfig.Bridge.JSONRPCEndpoint),
		txrelayer.WithWriter(logger.StandardWriter(&hclog.StandardLoggerOptions{})),
	)
	if err != nil {
		return err
	}

	b.checkpointManager = newCheckpointManager(
		wallet.NewEcdsaSigner(runtimeConfig.Key),
		runtimeConfig.GenesisConfig.Bridge.CheckpointManagerAddr,
		txRelayer,
		runtimeConfig.blockchain,
		runtimeConfig.polybftBackend,
		logger,
		runtimeConfig.State)
```

Reply:
Fix: https://github.com/Ethernal-Tech/blade/pull/47/commits/64db03d69359faf64096b84b14e11291939e97d4